lib/core/src/restore.rs

```diff
@@ -288,14 +347,20 @@ impl LiquidSdk {
 
                             // If neither is confirmed, this is an edge-case
                             (false, false) => {
-                                error!("Found unconfirmed lockup and refund txs while recovering data for Receive Swap {swap_id}");
+                                info!("Found unconfirmed lockup and refund txs while recovering data for Receive Swap {swap_id}");
                                 (None, None)
                             }
                         }
                     }
                 }
+                0 => {
+                    debug!(
+                        "No script history found while recovering data for Receive Swap {swap_id}"
+                    );
+                    (None, None)
+                }
                 n => {
-                    error!("Script history with unexpected length {n} found while recovering data for Receive Swap {swap_id}");
+                    info!("Script history with unexpected length {n} found while recovering data for Receive Swap {swap_id}");
```

Comment:
```suggestion
                    warn!("Script history with unexpected length {n} found while recovering data for Receive Swap {swap_id}");
```

Same comment as above about log severity, as this is an edge-case. Maybe `warn` is more appropriate?

Reply:
Actually, yeah we should revert to `error` I think for these unexpected length cases